### PR TITLE
add manifest action

### PR DESCRIPTION
**Pull request by ttata-cll**
(opening from a fork because I don't have write access to this repo)

Our team needs a workflow that will automatically build all nix packages in a given repo. We want to publish this as a re-usable action so we can share it across repos. https://smartcontract-it.atlassian.net/browse/INT2-413

### CODE REVIEW

**Comment by momentmaker** [comment]
nice! could you please add a changeset to this new action?

**Comment by ttata-cll** [comment]
> nice! could you please add a changeset to this new action?

sure! Are there instructions on how to do that? Sorry, I've never used that tool before

**Comment by momentmaker** [comment]
yeah you can do `pnpm i` first and then `pnpm changeset` at the root and check the action you just added :)

File: actions/manifest-build/action.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    required: false
+secrets:
+  github-token:
+    description: "GitHub auth token for private repos (if needed)"
+    required: false
+    default: ""
+  cache-token:
+      description: "Cachix auth token for private cache"
+      required: false
+      default: ""
```
Comment (ro-tex) [inline actions/manifest-build/action.yml:29]: Should we add an S3 option as well or is it too early?

Comment (ttata-cll) [reply actions/manifest-build/action.yml]: I think we should hold off and add that once we can test it e2e

**Review by momentmaker — APPROVED** [review]

**Pull request merged**